preprocessing/biotoolbox/structure_file_reader.py

```
@@ -1,142 +1,142 @@
+
+
+def build_structure_container_for_pdb(structure_data, chain_name):
+    # Test the data to see if this looks like a PDB or an mmCIF
+    tester = re.compile('^_', re.MULTILINE)
+    if len(tester.findall(structure_data)) == 0:
+        # PDB
+        parser_mode = 'pdb'
+    else:
+        parser_mode = 'cif'
+
+    temp = tempfile.TemporaryFile(mode='w+')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)